src/bids_validator/context.py

```diff
@@ -298,3 +303,128 @@ def from_file(cls, file: FileTree, schema: Namespace | None = None) -> t.Self:
             suffix=suffix,
             extension=extension,
         )
+
+
+@attrs.define
+class Context:
+    """A context object that creates context for file on access."""
+
+    file: FileTree
+    dataset: Dataset
+    subject: ctx.Subject
```

Comment:
Note that this can be `None`:

```suggestion
    subject: ctx.Subject | None
```